### DISCUSSION TITLE
Audio stream not work and stream was also closed when app did enter in the background

### DISCUSSION THREAD

**Discussion by luynv-2141** [HaishinKit/HaishinKit.swift#1194] [Q&A]
I tried version 1.4.5 and audio stream still not work when app did enter in background. I tried on code example and it not work too. Stream status is closed and auto reconnected when app was opened again.

I added some code like this but still does not work too

```
NotificationCenter.default.addObserver(self, selector: #selector(didEnterBackground(_:)), name: UIApplication.didEnterBackgroundNotification, object: nil)
NotificationCenter.default.addObserver(self, selector: #selector(didBecomeActive(_:)), name: UIApplication.didBecomeActiveNotification, object: nil)

   @objc
    private func didEnterBackground(_ notification: Notification) {
        rtmpStream.attachCamera(nil)
    }

    @objc
    private func didBecomeActive(_ notification: Notification) {
        rtmpStream.attachCamera(DeviceUtil.device(withPosition: currentPosition)) { error in
            logger.warn(error.description)
        }
    }
```

Device: iPhone XS Max
OS: 16.3.1

What I need is audio stream still work and stream still connected while stream is running. Please give me some supports. Thank you very much

**Comment by shogo4405** [comment]
Version 1.4.5 doesn't work with that code. Please use the main branch as it has been officially fixed there.

**Comment by luynv-2141** [comment]
@shogo4405 thank for your response. I tried main branch and audio worked, microphone symbol is displayed on top left corner of device when app enter in the background
But still have a problem, stream was closed in background and auto reconnected when app was opened again, I use https://castr.com/ to play m3u8 link and stream not play when my app entered in background. Is this a bug? I expect stream still play with only audio in this case

**Reply by shogo4405** [reply]
HaishinKit is under the BSD-3 License. I'm not providing support. If you can't do it, please give up. Or please fix yourself.
Repeating the same comment multiple times without resolution is just annoying. Thank you.

**Reply by luynv-2141** [reply]
Because audio stream work in background, but socket is not. so I just concerned that I miss some configured to use this feature perfectly
Sorry to make annoy, thank for your times
